### PR TITLE
#20 add register new external service sequence diagram

### LINKED ISSUE #20: Create sequence diagram of register new external service case

### PR DESCRIPTION
**Pull request by GiovaniBoff**
Closes #20 

image

### CODE REVIEW

**Review by nikolasFreitas — CHANGES_REQUESTED** [review]
Gio, could you be more specific on what type of error this could do, eg. more than one key for the same service or service already registered

**Review by evandroLutz — CHANGES_REQUESTED** [review]
I believe it would be valid to register the registration attempt process also in case of error, but returning an error when trying to register the service

File: docs/register-external-service.wsd [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    end box
+    database db as "Database"
+
+    client -> bff: Register new external service
+    activate bff
+    bff -> user: New external service registration
+    activate user
+    user -> user: Check registration inputs
+    alt Inputs are valid
+        user -> user: Generate credentials (JWT)\n [Access & Refresh]
```
Comment (nikolasFreitas) [inline docs/register-external-service.wsd:18]: the service must not generate the JWT on new service registration. The token is generated only after service login.

File: docs/register-external-service.wsd [hunk 2]
```diff
@@ -0,0 +1,33 @@
+@startuml
```
Comment (nikolasFreitas) [inline docs/register-external-service.wsd:1]: I think the User service must generate the credentials (account and password) for the external user. 
So the flow is: Admin register new service -> User service generate credentials -> Persist those and the service data into DB -> Return credentials with encrypted.

**Review by nikolasFreitas — CHANGES_REQUESTED** [review]

**Pull request merged**